### PR TITLE
files_external:list --importable-format documentation

### PR DESCRIPTION
**Pull request by phil-davis**
Fixes issue #2674
core code changes were merged in PR https://github.com/owncloud/core/pull/37513 for 10.5.0

Backport to 10.5 only.

### CODE REVIEW

**Comment by mmattel** [comment]
From core:
```
A new option to files_external:list has been added - -i or --importable-format - when this
is specified and a JSON output is requested, the items above are corrected in the JSON
to be in a format that can be used for import.
```
This reads for me that `-i` needs a proper output format (`json` or `json_pretty` ?) to be defined - but not plain which is used by default.

**Comment by phil-davis** [comment]
@mmattel any output format is valid. If used with a "normal" list then the text columns for `Storage`  and `Authentication Type` show the "technical" values rather than the "human readable" values.

See https://github.com/owncloud/core/pull/37513#issuecomment-642461919 - @jvillafanez also asked about this. It's good to see that reviewers are looking :)

**Comment by mmattel** [comment]
I have tested it just now.
I propose to add a text below the table like:

...
TIP: When using option `--importable-format`, consider using `--show-password` additionally. Else the user and/or admin has to change all mount passwords post importing manually to make a mount fully functional.
...

This is because if you miss `--show-password`, an import needs post work for either the user and/or the admin !

**Comment by phil-davis** [comment]
> This is because if you miss --show-password, an import needs post work for either the user and/or the admin !

OK, I will look at it. The "problem" here is that the new `--importable-format` option has only been added so that `files_external:export` can switch it on. The availability to someone manually doing `files_external:list --importable-format` is really an accidental side-effect. There is no intention for `files_external:list --importable-format` by itself to produce the output ready for `files_external:import` - for that there is the command `files_external:export` which puts together all the pieces that are needed.

**Comment by mmattel** [comment]
I think that the current implementation has a positive sideffect.
You have to mandatory add another option to make it fully work for importing.
This is a security measure ! :smile:
The only thing is, it must be documented.
Just thinking about, instead of a tip, we can write `NOTE:` and start with: For security measures you have to add...

**Comment by phil-davis** [comment]
@mmattel I wrote a "Note:" section

File: modules/admin_manual/pages/configuration/server/occ_command/files_external_commands.adoc [hunk 1]
```diff
@@ -68,6 +69,12 @@ be added as system mount.
 +----------+------------------+----------+
 ----
 
+NOTE: The `--importable-format` option helps make visible the technical mount settings.
```
Comment (mmattel) [inline modules/admin_manual/pages/configuration/server/occ_command/files_external_commands.adoc:72]: I am not a native speaker but this sounds better to me...
`helps make visible the technical mount settings.` --> `helps to make the technical mount settings visible.`

Comment (phil-davis) [reply modules/admin_manual/pages/configuration/server/occ_command/files_external_commands.adoc]: I should read what I write! Fixed.

**Comment by phil-davis** [comment]
Backport to 10.5 is PR #2678

**Pull request merged**